docassemble/AssemblyLine/data/questions/al_saved_sessions.yml

```diff
@@ -167,31 +174,32 @@ code: |
     #'user_started_case',
   ]
 ---
+code: |
+  def file_like(obj):
+    return isinstance(obj, DAFile) or isinstance(obj, DAFileCollection) or isinstance(obj, DAFileList) or isinstance(obj, ALDocument) or isinstance(obj, ALDocumentBundle)
+---
+need:
+  - file_like
 # For save snapshot
 code: |
   filtered_vars_tmp = all_variables(simplify=False)
   simple_filtered_vars_tmp = all_variables()
   
   for var in al_sessions_variables_to_remove:
     filtered_vars_tmp.pop(var, None)
-    simple_filtered_vars_tmp.pop(var, None)
-  
-  # Make all top-level DAFiles and DAFileCollections public
-  # TODO: is it worth making this recursive and traversing all attributes?
-  for var in filtered_vars_tmp:
-    if isinstance(var, DAFile) or isinstance(var, DAFileCollection) or isinstance(var, DAFileList):
-      try:
-        var.set_attributes(private=False)
-      except:
-        pass
+    simple_filtered_vars_tmp.pop(var, None)   
   
   al_simple_filtered_vars = simple_filtered_vars_tmp
```

Comment:
1. Do we need `simple_filtered_vars_tmp`? If we use `al_simple_filtered_vars` to begin with, it will save a few lines.
2. I meant to ask this question on Monday's call but we didn't get to it. Most packages are unique, why do we include saved sessions from outside the current interview? Is it intentional? 

1. The current session is listed on the source session. Should it be removed?

1.  Remove the down arrow as shown in the image since it's redundant and doesn't do anything?

![image](https://user-images.githubusercontent.com/20231649/157524753-2924a2e6-6d33-4dde-85fb-13bc5c8c58de.png)

Reply:
> 1. Do we need `simple_filtered_vars_tmp`? If we use `al_simple_filtered_vars` to begin with, it will save a few lines.

Yes we do, it's to avoid Docassemble stopping running the code block too early. 

>     2. I meant to ask this question on Monday's call but we didn't get to it. Most packages are unique, why do we include saved sessions from outside the current interview? Is it intentional?

Yes. This feature is eventually designed to let you copy answers from one interview into another interview. It is true that it will only work for fields that have the same name.

>     3. The current session is listed on the source session. Should it be removed?

Yes. I'll make changes like that in a new PR, though. I want to make it a lot simpler overall, actually.

>     4. Remove the down arrow as shown in the image since it's redundant and doesn't do anything?
> 
> 
> ![image](https://user-images.githubusercontent.com/20231649/157524753-2924a2e6-6d33-4dde-85fb-13bc5c8c58de.png)

That's a docassemble CSS bug for the combobox input that I've also noticed recently.